build.gradle

```diff
@@ -45,17 +45,15 @@ tasks.withType(JavaCompile).configureEach {
 	// see http://yodaconditions.net/blog/fix-for-java-file-encoding-problems-with-gradle.html
 	// If Javadoc is generated, this must be specified in that task too.
 	it.options.encoding = "UTF-8"
+}
 
+java {
 	// The Minecraft launcher currently installs Java 8 for users, so your mod probably wants to target Java 8 too
-	// JDK 9 introduced a new way of specifying this that will make sure no newer classes or methods are used.
-	// We'll use that if it's available, but otherwise we'll use the older option.
-	def targetVersion = 8
-	if (JavaVersion.current().isJava9Compatible()) {
-		 it.options.release = targetVersion
+	// This ensures that no newer classes, methods, or language features are used.
+	toolchain {
+		languageVersion = JavaLanguageVersion.of(8)
```

Comment:
Doesnt this download and use java 8? The idea before was to build on the newer JDKs while ensuring that it will run on 8.

Setting the release compiler arg does the same thing without actaully needing to run on java 8. I want to move further away from j8 this seems to be moving closer to it by not even ensuing stuff builds on newer versions.

Reply:
Urf, wasn't aware of this, would make more sense to add something in loom to specify the target JVM version then instead of having an obscure option manipulation.

Reply:
Its not so obscure, propper support for it was added in gradle 6.6 before it was a matter of setting a compiler argument. 

https://docs.gradle.org/6.6/release-notes.html#javacompile-release

It does look like we could possibly make it a little neater.

Reply:
When we first added this, sourceCompatibility/target was removed but I found its best to keep those in as well to ensure IDEs use the correct target version.

Reply:
imo, Gradle should have added a proper field for that in the `java` block like it does for `sourceCompatibility` and `targetCompatibility`.